web/components/roadmap/MiniRoadmap.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { ReactElement, useCallback, useEffect, useState } from "react";
+import { useRoadmap } from "@/lib/data-hooks/useRoadmap";
+import { VerticalStepIndicator } from "@/components/njwds-extended/VerticalStepIndicator";
+import { MiniRoadmapTask } from "@/components/roadmap/MiniRoadmapTask";
+import { Icon } from "@/components/njwds/Icon";
+import analytics from "@/lib/utils/analytics";
+import { useUserData } from "@/lib/data-hooks/useUserData";
+import { isStepCompleted } from "@/lib/utils/helpers";
+import { SectionType } from "@/lib/types/types";
+import { SectionDefaults } from "@/display-content/roadmap/RoadmapDefaults";
+import { Accordion, AccordionSummary, AccordionDetails } from "@mui/material";
+import { getSectionNames } from "@/lib/utils/helpers";
+
+interface Props {
+  activeTaskId: string;
+  onTaskClick?: () => void;
+}
+
+export const MiniRoadmap = (props: Props): ReactElement => {
+  const { roadmap } = useRoadmap();
+  const getActiveStepNumber = useCallback(
+    () =>
+      roadmap?.steps.find((step) => step.tasks.map((it) => it.id).includes(props.activeTaskId))?.step_number,
+    [props.activeTaskId, roadmap?.steps]
+  );
+  const [activeStepNumber, setActiveStepNumber] = useState<number | undefined>(getActiveStepNumber());
+  const [openSteps, setOpenSteps] = useState<number[]>([]);
+  const { userData, update } = useUserData();
+  const roadmapExists = !!roadmap;
+  const roadmapSections = getSectionNames(roadmap);
+
+  useEffect(() => {
+    const activeStep = getActiveStepNumber();
+    setActiveStepNumber(activeStep);
+    setOpenSteps(activeStep ? [activeStep] : []);
+  }, [props.activeTaskId, roadmapExists, getActiveStepNumber]);
+
+  const toggleStep = (stepNumber: number): void => {
+    analytics.event.task_mini_roadmap_step.click.expand_contract();
+    if (openSteps.includes(stepNumber)) {
+      setOpenSteps(openSteps.filter((number) => number !== stepNumber));
+    } else {
+      setOpenSteps([...openSteps, stepNumber]);
+    }
+  };
+
+  const handleAccordionStateChange = async (
+    sectionType: SectionType,
+    expanded: boolean | undefined
+  ): Promise<void> => {
+    const openRoadMapSections = userData?.preferences.roadmapOpenSections;
```

Comment:
let's try to keep variable naming consistent. if it's spelled `roadmapOpenSections` , it's kind of confusing to rename that variable to `openRoadMapSections`
- the switching around "open" and "roadmap" is confusing, why are we changing order? is it supposed to convey something about the value of the variable?
- if it's spelled "Roadmap" we should keep the same spelling, not change to "RoadMap"

Reply:
Renamed openRoadMapSections to roadmapOpenSections for consistency.